src/model/resources/fusion/fusion.jl

```diff
@@ -0,0 +1,545 @@
+    end
+end
+
+@doc raw"""
+    _fusion_dwell_avoided_operation(dwell_time::Float64,
+                                    ePulseStart::AffExpr)
+
+    dwell_time in fractions of a timestep
+    ePulseStart is the number of MW starting. Typically component_size * vPulseStart
+"""
+function _fusion_dwell_avoided_operation(dwell_time::Float64, ePulseStart::AffExpr)
+    return dwell_time * ePulseStart
+end
+
+function fusion_parasitic_power!(
+        EP::Model,
+        inputs::Dict,
+        resource_component,
+        r_id::Int,
+        reactor::FusionReactorData,
+        component_capacity::Symbol
+)
+    T = inputs["T"]
+
+    component_size = reactor.component_size
+    parasitic_passive_fraction = reactor.parasitic_passive_fraction
+    parasitic_active_fraction = reactor.parasitic_active_fraction
+    parasitic_start_energy_fraction = reactor.parasitic_start_energy_fraction
+    pulse_start_power_fraction = reactor.pulse_start_power_fraction
+    dwell_time = reactor.dwell_time
+    η = reactor.eff_down
+
+    from_model(f::Function) = EP[Symbol(f(resource_component))]
+
+    pulsestart = component_size * from_model(fusion_pulse_start_name)
+    underway = component_size * from_model(fusion_pulse_underway_name)
+
+    capacity = EP[component_capacity][r_id]
+
+    passive = Symbol(fusion_parasitic_passive_name(resource_component))
+    active = Symbol(fusion_parasitic_active_name(resource_component))
+    start_energy = Symbol(fusion_parasitic_start_energy_name(resource_component))
+    pulse_start_power = Symbol(fusion_pulse_start_power_name(resource_component))
+
+    # Passive recirculating power, depending on built capacity
+    EP[passive] = @expression(EP, [t in 1:T], capacity*η*parasitic_passive_fraction)
+
+    # Active recirculating power, depending on whether a pulse is underway
+    EP[active] = @expression(EP,
+        [t in 1:T],
+        η*
+        parasitic_active_fraction*
+        (underway[t]-pulsestart[t] * dwell_time))
+    # Startup energy, taken from the grid every time the core starts up
+    EP[start_energy] = @expression(EP,
+        [t in 1:T],
+        pulsestart[t]*η*parasitic_start_energy_fraction)
+
+    # Startup power, required margin on the grid when the core starts
+    EP[pulse_start_power] = @expression(EP,
+        [t in 1:T],
+        pulsestart[t]*η*pulse_start_power_fraction)
+    union!(inputs[FUSION_PULSE_START_POWER], (pulse_start_power,))
+end
+
+function fusion_total_parasitic_power!(
+        EP::Model,
+        inputs::Dict,
+        resource_component,
+        r_id::Int
+)
+    from_model(f::Function) = EP[Symbol(f(resource_component))]
+
+    ePassive = from_model(fusion_parasitic_passive_name)
+    eActive = from_model(fusion_parasitic_active_name)
+    eStartEnergy = from_model(fusion_parasitic_start_energy_name)
+
+    total_parasitic = Symbol(fusion_parasitic_total_name(resource_component))
+
+    EP[total_parasitic] = ePassive + eActive + eStartEnergy
```

Comment:
If possible, create expressions inside JuMP macros ([Ref](https://jump.dev/JuMP.jl/stable/tutorials/getting_started/performance_tips/#Use-macros-to-build-expressions))

Reply:
done.